### ISSUE TITLE
Mono Cross-compilation doesn't work

### ISSUE THREAD

**Issue by lmtr0** [picoe/Eto#1802]
## Expected Behavior
Mono Compile the app and the app runs

## Actual Behavior
Mono compile the app but it doens't open the app

## Steps to Reproduce the Problem

  1. Create The Template app `dotnet new etoapp`
  2. Remove the netcoreapp3.1 of the TargetFrameworks in the project.csproj
  3. run msbuild in the Wpf directory
  4. run mono APP.exe 

## Code that Demonstrates the Problem

```
using System;
using Eto.Forms;

namespace HelloEto.Wpf
{
	class MainClass
	{
		[STAThread]
		public static void Main(string[] args)
		{
			new Application(Eto.Platforms.Wpf).Run(new MainForm());
		}
	}
}
```

## Specifications

  - Version: 5.8.14-arch1-1
  - Platform(s): WPF, WinForms
  - Operating System(s):Archlinux
 ## aditional Help
I am trying to compile to windows using mono because dotnet doesn't made it. Is there any runtime that supports what I am trying to do?

**Comment by lmtr0** [comment]
Mono APP.exe throws:
```Console
Can't find custom attr constructor image: /home/myuser/repo/HelloEto/HelloEto.Wpf/bin/Debug/net472/Eto.Wpf.dll mtoken: 0x0a000005 due to: Could not load file or assembly 'PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies.

Unhandled Exception:
System.InvalidOperationException: Error creating instance of platform type 'Eto.Wpf.Platform, Eto.Wpf' ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.TypeInitializationException: The type initializer for 'Eto.Wpf.Platform' threw an exception. ---> System.BadImageFormatException: Could not resolve field token 0x04000378, due to: Could not load type of field 'Eto.Wpf.Platform+<>c:<>9__8_1' (1) due to: Could not load file or assembly 'PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. assembly:/home/myuser/repo/HelloEto/HelloEto.Wpf/bin/Debug/net472/Eto.Wpf.dll type:<>c member:(null)
   --- End of inner exception stack trace ---
  at (wrapper managed-to-native) System.Reflection.RuntimeConstructorInfo.InternalInvoke(System.Reflection.RuntimeConstructorInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x00005] in <671ef0784b04474c83bb500a849492f2>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x0001a] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.RuntimeType.CreateInstanceMono (System.Boolean nonPublic, System.Boolean wrapExceptions) [0x00095] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.RuntimeType.CreateInstanceSlow (System.Boolean publicOnly, System.Boolean wrapExceptions, System.Boolean skipCheckThis, System.Boolean fillCache) [0x00009] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.RuntimeType.CreateInstanceDefaultCtor (System.Boolean publicOnly, System.Boolean skipCheckThis, System.Boolean fillCache, System.Boolean wrapExceptions, System.Threading.StackCrawlMark& stackMark) [0x00027] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic, System.Boolean wrapExceptions) [0x00020] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic) [0x00000] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Activator.CreateInstance (System.Type type) [0x00000] in <671ef0784b04474c83bb500a849492f2>:0 
  at Eto.Platform.Get (System.String platformType, System.Boolean allowNull, System.Collections.Generic.List`1[T] errors) [0x0002c] in <620a34681a66479b88c8264123897a86>:0 
   --- End of inner exception stack trace ---
  at Eto.Platform.Get (System.String platformType, System.Boolean allowNull, System.Collections.Generic.List`1[T] errors) [0x00096] in <620a34681a66479b88c8264123897a86>:0 
  at Eto.Forms.Application..ctor (System.String platformType) [0x00000] in <620a34681a66479b88c8264123897a86>:0 
  at HelloEto.Wpf.MainClass.Main (System.String[] args) [0x00001] in <cf93476c28934223ab00432ab677a436>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: Error creating instance of platform type 'Eto.Wpf.Platform, Eto.Wpf' ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.TypeInitializationException: The type initializer for 'Eto.Wpf.Platform' threw an exception. ---> System.BadImageFormatException: Could not resolve field token 0x04000378, due to: Could not load type of field 'Eto.Wpf.Platform+<>c:<>9__8_1' (1) due to: Could not load file or assembly 'PresentationFramework, Version=4.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' or one of its dependencies. assembly:/home/myuser/repo/HelloEto/HelloEto.Wpf/bin/Debug/net472/Eto.Wpf.dll type:<>c member:(null)
   --- End of inner exception stack trace ---
  at (wrapper managed-to-native) System.Reflection.RuntimeConstructorInfo.InternalInvoke(System.Reflection.RuntimeConstructorInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x00005] in <671ef0784b04474c83bb500a849492f2>:0 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeConstructorInfo.InternalInvoke (System.Object obj, System.Object[] parameters, System.Boolean wrapExceptions) [0x0001a] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.RuntimeType.CreateInstanceMono (System.Boolean nonPublic, System.Boolean wrapExceptions) [0x00095] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.RuntimeType.CreateInstanceSlow (System.Boolean publicOnly, System.Boolean wrapExceptions, System.Boolean skipCheckThis, System.Boolean fillCache) [0x00009] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.RuntimeType.CreateInstanceDefaultCtor (System.Boolean publicOnly, System.Boolean skipCheckThis, System.Boolean fillCache, System.Boolean wrapExceptions, System.Threading.StackCrawlMark& stackMark) [0x00027] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic, System.Boolean wrapExceptions) [0x00020] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Activator.CreateInstance (System.Type type, System.Boolean nonPublic) [0x00000] in <671ef0784b04474c83bb500a849492f2>:0 
  at System.Activator.CreateInstance (System.Type type) [0x00000] in <671ef0784b04474c83bb500a849492f2>:0 
  at Eto.Platform.Get (System.String platformType, System.Boolean allowNull, System.Collections.Generic.List`1[T] errors) [0x0002c] in <620a34681a66479b88c8264123897a86>:0 
   --- End of inner exception stack trace ---
  at Eto.Platform.Get (System.String platformType, System.Boolean allowNull, System.Collections.Generic.List`1[T] errors) [0x00096] in <620a34681a66479b88c8264123897a86>:0 
  at Eto.Forms.Application..ctor (System.String platformType) [0x00000] in <620a34681a66479b88c8264123897a86>:0 
  at HelloEto.Wpf.MainClass.Main (System.String[] args) [0x00001] in <cf93476c28934223ab00432ab677a436>:0 
```

**Comment by cwensley** [comment]
Hi @litch0,

WPF can only run on Windows.  It appears that the cross _compiling_ worked, just not running.  If you run that program on Windows, it should work.

WinForms _might_ work on Mono, but I don't support that scenario.

I recommend to use GTK on Linux, WPF or WinForms on Windows, and Mac64 or XamMac2 on Mac.

**Comment by lmtr0** [comment]
thx

**Comment by lmtr0** [comment]
winforms woks
![image](https://user-images.githubusercontent.com/57605930/96274935-7a417f80-0fa7-11eb-9905-37f78657b496.png)

**Comment by greatawesome** [comment]
Semi-related to this thread, I did some testing of the Eto.WinForms backend on non-Windows platforms. I know it isn't a supported scenario, but I was curious about embedding WinForms in Eto and Eto in WinForms scenarios, as either option could support a staged migration of a WinForms app to Eto. Once the migration was complete and all the WinForms bits are gone, the Eto backend could be flipped to the native GTk/Mac version. 

Here is what I found running the Eto embedding test apps:

**Ubuntu 16.04 LTS, Mono 4.0**
Embedding Winforms in Eto works
Embedding Eto in WinForms DOES NOT work, gives a blank window

**Mac OS Mojave 10.14.6, Mono 6.12 32 bit**
Embedding Winforms in Eto works
Embedding Eto in Winforms works
Will not run in 64 bit due to Mono’s reliance on 32bit Carbon calls for WinForms

I didn't dig into why embedding Eto in WinForms failed on Linux. For all I know it'd be a simple fix, or maybe it is just my distro. If not, it'd suggest any WinForms to Eto porting strategy would require converting the main application to an Eto app, and embedding your WinForms components into it.

MacOS newer than 10.15 will be a problem since it won't run 32 bit code. As of now, the only way to get WinForms working in 64bit would be to try the mac-playground fork of WinForms.
